pkgs/by-name/vs/vs-dfttest2/package.nix

```diff
@@ -0,0 +1,78 @@
+{
+  lib,
+  gcc12Stdenv,
+  fetchFromGitHub,
+  cmake,
+  pkg-config,
+  cudaPackages_11_8,
+  rocmPackages,
+  vapoursynth,
+  zimg,
+  cudaSupport ? false,
+  rocmSupport ? false,
+  genericVector ? false,
+}:
+
+# Based on project's CI
+gcc12Stdenv.mkDerivation (finalAttrs: {
+  pname = "vs-dfttest2";
+  version = "7";
+
+  src = fetchFromGitHub {
+    fetchSubmodules = true;
```

Comment:
You could use a separately packaged `vectorclass` instead of fetching submodules.